### ISSUE TITLE
Resolve suppressed Coroutines related `DEPRECATION` warning for `ReviewModerationHandlerTests`

### ISSUE THREAD

**Issue by ParaskP7** [woocommerce/woocommerce-android#6899]
As part of #6902 and this [commit](https://github.com/woocommerce/woocommerce-android/pull/6902/commits/4a6304d4c8f3e6644bc3e42f0e88c99189a451bd), the `ReviewModerationHandlerTests` test and all its warnings got suppressed with `DEPRECATION`.

The warnings occur because this `ReviewModerationHandlerTests` test suite is the only such suite that still uses the old way of testing with Coroutines. As part of the [Coroutine 1.6 PR](https://github.com/woocommerce/woocommerce-android/pull/6292), migrating this test suite to the new way didn't happen as different approaches using both 'Standard' and 'Unconfined' dispatchers didn't work, 3 tests were always failing. One would need to remove this figure out how to use the new way of testing Coroutines on this `ReviewModerationHandlerTests`, maybe even rewriting it if need be.

**Comment by ParaskP7** [comment]
👋 @malinajirka after trying to run a specific unit test I wrote with [Konsist](https://docs.konsist.lemonappdev.com) (me working on [this](https://github.com/Automattic/apps-infra-plans/issues/157)) I got a build failure on that [ReviewModerationHandlerTests](https://github.com/woocommerce/woocommerce-android/blob/trunk/WooCommerce/src/test/kotlin/com/woocommerce/android/ui/reviews/ReviewModerationHandlerTests.kt#L38) test. Then, I noticed this [TODO](https://github.com/woocommerce/woocommerce-android/blob/689bd21c28405f28271c99ff97269942e4a50ae6/WooCommerce/src/test/kotlin/com/woocommerce/android/ui/reviews/ReviewModerationHandlerTests.kt#L2) with your name on it, followed it and remembered this issue here... 🤔

Maybe it is high-time we deal with it, wdyt? 🙏

**Comment by malinajirka** [comment]
I agree it'd be great, having said that, I'm unlikely to have time for this in the next month or so. I need to move POS Coupons as much as I can before my AFK in April, and I'll be gone for a couple weeks then. I can put this on my list for the next hackweek.

**Comment by ParaskP7** [comment]
Awesome @malinajirka , thanks for adding it to your TODO list no matter when that could be tackled on, it is good we don't forget about it.

FYI: I'll also add it to our [Maintenance Backlog](https://github.com/Automattic/apps-infra-plans/issues/191), just to keep track of it, and who knows, maybe we'll deal with it first.
